llvm/include/llvm/LTO/legacy/ThinLTOCodeGenerator.h

```diff
@@ -125,11 +148,50 @@ class ThinLTOCodeGenerator {
   struct CachingOptions {
     std::string Path;                    // Path to the cache, empty to disable.
     CachePruningPolicy Policy;
+    bool UseCAS = false;
+    std::unique_ptr<cas::ObjectStore> CAS;
+    std::unique_ptr<cas::ActionCache> Cache;
   };
 
   /// Provide a path to a directory where to store the cached files for
   /// incremental build.
-  void setCacheDir(std::string Path) { CacheOptions.Path = std::move(Path); }
+  Error setCacheDir(std::string Path) {
+    StringRef SocketPath = Path;
+    // The environment overwrites the option parameter.
+    if (auto EnvCachePath =
+            sys::Process::GetEnv("LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH")) {
```

Comment:
I would prefer we move this logic out of the C++ library code, e.g. to `thinlto_codegen_set_cache_dir` library function and llvm-lto tool.